src/components/TweetList.js

```diff
@@ -11,7 +11,31 @@ const defaultProps = {
 };
 
 function renderTweets(tweets) {
-  return tweets.map(tweet => <Tweet tweet={tweet} key={tweet.id} />);
+
+  tweets.forEach(tweet => {
+    let imageSmall;
+
+    if(tweet['entities']['media'] !== undefined){
+      console.log(tweet['entities']);
+      imageSmall = tweet['entities']['media'][0]['media_url'] + ':small';
+      Object.assign(tweet, {imageSmall: imageSmall});
+    }
+
+    let link;
+
+    if(/http\S*/.test(tweet['text'])){
+      link = tweet['text'].match(/http\S*/g).join('')
```

Comment:
There is an easier way, the tweet data contains urls in the entities. You just need to search for these strings. Reg ex searches should be avoided if possible doing a substring check is quicker. Also this will cause a problem if there are multiple urls in the text. This would result in 2 or more urls being joined together and end up with a broken link.